Sources/ArgumentParser/Parsable Properties/CompletionKind.swift

```diff
@@ -9,56 +9,149 @@
   }
 
-  /// Complete file names.
+  /// The completion candidates include directory & file names, the latter
+  /// filtered by the given list of extensions.
+  ///
+  /// If the given list of extensions is empty, then file names are not
+  /// filtered.
+  ///
+  /// Given file extensions must not include the `.` initial extension
+  /// separator.
+  ///
+  /// Given file extensions are parsed by the requesting shell as globs; Swift
+  /// Argument Parser does not perform any escaping or quoting.
+  ///
+  /// The directory/file filter & the given list of extensions are included in a
+  /// completion script when it is generated.
   public static func file(extensions: [String] = []) -> CompletionKind {
     CompletionKind(kind: .file(extensions: extensions))
   }
 
-  /// Complete directory names.
+  /// The completion candidates are directory names.
+  ///
+  /// The directory filter is included in a completion script when it is
+  /// generated.
   public static var directory: CompletionKind {
     CompletionKind(kind: .directory)
   }
 
-  /// Call the given shell command to generate completions.
+  /// The completion candidates are specified by the stdout output of the given
+  /// string run as a shell command when a user requests completions.
+  ///
+  /// Swift Argument Parser does not perform any escaping or quoting on the
+  /// given shell command.
+  ///
+  /// The given shell command is included in a completion script when it is
+  /// generated.
   public static func shellCommand(_ command: String) -> CompletionKind {
     CompletionKind(kind: .shellCommand(command))
   }
 
-  /// Generate completions using the given closure.
+  /// The completion candidates are the strings in the array returned by the
+  /// given closure when it is run when a user requests completions.
+  ///
+  /// Completion candidates are interpreted by the requesting shell as literals.
+  /// They must be neither escaped nor quoted; Swift Argument Parser escapes or
+  /// quotes them as necessary for the requesting shell.
+  ///
+  /// The given closure is evaluated after a user invokes completion in their
+  /// shell (normally by pressing TAB); it is not evaluated when a completion
+  /// script is generated.
+  ///
+  /// The array of strings passed to the given closure contains all the shell
+  /// words in the command line for the current command at completion
+  /// invocation; this is exclusive of words for prior or subsequent commands or
+  /// pipes, but inclusive of redirects & any other command line elements. Each
+  /// word is its own element in the argument array; they appear in the same
+  /// order as in the command line. Note that shell words may contain spaces if
+  /// they are escaped or quoted.
+  ///
+  /// Shell words are passed to Swift verbatim, not unquoted. e.g., the
+  /// representation in Swift of the shell word `"abc\\""def"` would be exactly
+  /// the same, including the double quotes & the double backslash.
```

Comment:
I think I know what you mean by "not unquoted", but the adjacent double negative is hard to parse. Can you expand this paragraph a bit? Let's also spell out "For example" instead of "e.g.".